Xamarin.Essentials/Share/Share.shared.cs

```diff
@@ -15,11 +14,28 @@ public static partial class Share
         public static Task RequestAsync(string text, string title) =>
             RequestAsync(new ShareTextRequest(text, title));
 
-        public static Task RequestAsync(ShareTextRequest request) =>
-            PlatformRequestAsync(request);
+        public static Task RequestAsync(ShareTextRequest request)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+
+            if (string.IsNullOrEmpty(request.Text) && string.IsNullOrEmpty(request.Uri))
+                throw new ArgumentNullException(nameof(request.Text));
```

Comment:
This will possibly confuse people if they pass a sharerequest with a null uri, they would get the wrong nameof expression. Splitting this into two if statements removes that problem

Reply:
Good catch.

Maybe this should not be an `ArgumentNullException` since the argument was not null. Maybe a `ArgumentException` with a better message? "Both the Text and Uri were invalid. Make sure to xyz." or something?

Reply:
In an ideal world I would love to prevent this from occuring at all instead of having a developer have  to investigate which parameters are valid. I know it would be a breaking change, but changing shareRequest to have readonly properties and having only constructors exposed which set required fields. This is the solution imho, as being able to construct objects which always fail if not all parameters are set is not very elegant